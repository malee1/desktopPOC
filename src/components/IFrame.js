import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { useSelector } from 'react-redux';
// import defaultUrl from '../constants/defaultUrl';

const useStyles = makeStyles({
  root: {
    backgroundColor: 'white',
    flexGrow: 1,
    height: '100%',
    width: '100%'
  }
});

export default function IFrame() {
  const classes = useStyles();

  const url = useSelector((state) => state.iFrame.iFrameUrl);

  return (
    <div className={classes.root} data-testid="iframe">
      <iframe
        title="title"
        src={url}
        allowFullScreen
        allow="encryted-media"
        height="100%"
        width="100%"
      />
    </div>
  );
}
