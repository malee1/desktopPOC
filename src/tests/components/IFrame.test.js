import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import IFrame from '../../components/IFrame';

const mockStore = configureMockStore();
const store = mockStore({
  iFrame: {
    iFrameUrl: 'https://freeklime.co.uk'
  }
});

describe('the iFrame renders', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <IFrame />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('there is one component', () => {
    const { getAllByTestId } = render(
      <Provider store={store}>
        <IFrame />
      </Provider>
    );
    const container = getAllByTestId('iframe');
    expect(container.length).toBe(1);
  });
});
