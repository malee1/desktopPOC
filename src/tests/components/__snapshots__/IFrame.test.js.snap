// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the iFrame renders matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="iframe"
>
  <iframe
    allow="encryted-media"
    allowFullScreen={true}
    height="100%"
    src="https://freeklime.co.uk"
    title="title"
    width="100%"
  />
</div>
`;
